src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -137,21 +149,21 @@ private void CallContractInternal(UInt160 contractHash, string method, Array arg
             if (md is null) throw new InvalidOperationException($"Method {method} Does Not Exist In Contract {contractHash}");
 
             ContractManifest currentManifest = Snapshot.Contracts.TryGet(CurrentScriptHash)?.Manifest;
```

Comment:
This BTW can now be optimized for entry scripts as no valid entry script could have a hash of deployed contract (as long as we trust our hashes).